physlr/physlr.py

```diff
@@ -1201,6 +1207,27 @@ def physlr_overlap(self):
             (u, v) for bxs in progress(mxtobxs.values()) for u, v in itertools.combinations(bxs, 2))
         print(int(timeit.default_timer() - t0), "Loaded", len(edges), "edges", file=sys.stderr)
 
+        edge_list = list(edges)
+        neighbour_min = defaultdict(list)
+
+        for edge in edge_list:
+            neighbour_min[edge[0]] += bxtomxs[edge[1]]
+            neighbour_min[edge[1]] += bxtomxs[edge[0]]
+        for i in progress(bxtomxs):
+            neighbour_min[i] = set(neighbour_min[i])
+        print(
+            int(timeit.default_timer() - t0),
+            "Found", len(neighbour_min), "edges weights", file=sys.stderr)
+        Physlr.neighbour_min = neighbour_min
+        with multiprocessing.Pool(self.args.threads) as pool:
+            edges2 = dict(x for l in pool.map(self.physlr_overlap_minimizer_intersection,
+                                              progress(edge_list),
+                                              chunksize=100) for x in l)
+        edges = edges2
+        print(
+            int(timeit.default_timer() - t0),
+            "Recalculated", len(edges), "edges weights", file=sys.stderr)
+
         for (u, v), n in progress(edges.items()):
```

Comment:
So currently we are not doing the update of the edge weights on the graph right?

Reply:
It made more sense to me to do update the edge weight and add the edges to the graph